controllers/infrastructure/byomachine_controller.go

```diff
@@ -259,27 +259,25 @@ func (r *ByoMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 		r.Recorder.Eventf(machineScope.ByoMachine, corev1.EventTypeNormal, "ByoHostAttachSucceeded", "Attached ByoHost %s", machineScope.ByoHost.Name)
 	}
 
-	if machineScope.ByoMachine.Spec.ProviderID == "" {
-		logger.Info("Updating Node with ProviderID")
-		remoteClient, err := r.getRemoteClient(ctx, machineScope.ByoMachine)
-		if err != nil {
-			logger.Error(err, "failed to get remote client")
-			return ctrl.Result{}, err
-		}
-
-		providerID, err := r.setNodeProviderID(ctx, remoteClient, machineScope.ByoHost)
-		if err != nil {
-			logger.Error(err, "failed to set node providerID")
-			r.Recorder.Eventf(machineScope.ByoMachine, corev1.EventTypeWarning, "SetNodeProviderFailed", "Node %s does not exist", machineScope.ByoHost.Name)
-			return ctrl.Result{}, err
-		}
+	logger.Info("Updating Node with ProviderID")
+	remoteClient, err := r.getRemoteClient(ctx, machineScope.ByoMachine)
```

Comment:
@huchen2021 
I haven't completely gone through the PR, but I assumed this PR was only to add tests. Why the changes in `byomachine_controller`?

Reply:
There are two code changes as followed:
1) One is for removing the if line, it introduced by @dharmjit. The original behavior: " If machineScope.ByoMachine.Spec.ProviderID is empty, don't call setNodeProviderID function". This breaks code of issue-290. Even machineScope.ByoMachine.Spec.ProviderID is empty, it may has invalid value, and it shouldn't just let it go. In the code of issue-290, inside of function setNodeProviderID, it will check the value of machineScope.ByoMachine.Spec.ProviderID and do according process.

2) The other is to Capitalize the first letter of providerIDSuffixLength to export it, then controll_test.go can use the variable.

Reply:
> dharmjit and I almost check in the code in the same time

@huchen2021, I am not sure which 2 changes were pushed at the same time. The `if` check is not introduced lately.

Reply:
> > dharmjit and I almost check in the code in the same time
> 
> @huchen2021, I am not sure which 2 changes were pushed at the same time. The `if` check is not introduced lately.

Sorry, i read it wrong from "5 months ago" to "5 days ago".
Anyway, this if check should be removed.
Sorry for that again.